### PR TITLE
[CALCITE-3775] Implicit lookup methods in SimpleCalciteSchema ignore case sensitivity parameter (Alon Eldar)

### PR DESCRIPTION
**Pull request by AlonEldar89**
rebased original  PR #1792 from Feb2020.

### CODE REVIEW

**Review by NobiGo — APPROVED** [review]
LGTM

File: core/src/test/java/org/apache/calcite/test/JdbcTest.java [hunk 1]
```diff
@@ -7716,6 +7760,8 @@ private void checkGetTimestamp(Connection con) throws SQLException {
     assertThat.query(query).returns("id=2008-03-31 12:23:34\nid=2018-02-03 00:00:00\n");
   }
 
+
+
```
[the remark this answers is not in the document]
Comment (DonnyZone) [reply core/src/test/java/org/apache/calcite/test/JdbcTest.java:7764]: Agree, we need to remove the extra empty row.

**Comment by DonnyZone** [comment]
LGTM

**Review by DonnyZone — APPROVED** [review]

File: core/src/main/java/org/apache/calcite/jdbc/SimpleCalciteSchema.java [hunk 2]
```diff
@@ -74,33 +76,73 @@ private SimpleCalciteSchema(@Nullable CalciteSchema parent,
...
+    // Fall through: Set iteration
+    for (String candidate: candidates) {
+      if (candidate.equalsIgnoreCase(name)) {
+        return candidate;
+      }
+    }
+    return null;
+  }
+
+
```
Comment (DonnyZone) [inline core/src/main/java/org/apache/calcite/jdbc/SimpleCalciteSchema.java:103]: Nitpicking: extra line

**Comment by DonnyZone** [comment]
Hi, @AlonEldar89, when you remove the extra line. Could you squash the commits together? The "squash and merge" is not enabled in Calcite. So that I can merge it quickly just through the Github page.

**Comment by AlonEldar89** [comment]
> Hi, @AlonEldar89, when you remove the extra line, could you squash the commits together? The "squash and merge" is not enabled in Calcite. So that I can merge it quickly just through the Github page. Thanks!

Done.

**Comment by DonnyZone** [comment]
The style check failed.
`* What went wrong:
Execution failed for task ':core:autostyleJavaCheck'.
> The following files have format violations:
    core\src\main\java\org\apache\calcite\jdbc\SimpleCalciteSchema.java
      @@ -99,7 +99,7 @@
           }
           return null;
         }
      -  \r\n
      +\r\n
         @Override protected @Nullable CalciteSchema getImplicitSubSchema(String schemaName,
             boolean caseSensitive) {
           // Check implicit schemas.
  Run './gradlew autostyleApply' to fix the violations.
`

**Pull request merged**